### PR TITLE
Add GhoStableDebtToken implementation

### LINKED ISSUE #316: Nonexistent StableDebtToken for Gho
GHO needs to have a StableDebtToken implementation so it can be used in the Aave listing.

It can use the canonical standard StableDebtToken implementation, or a custom implementation that reverts on mint/burn calls. Both of them work well because GHO is not intended to be activated for stable borrowing, but the latter feels safer so even an error in the gov process won't impact GHO functionality and design.

### PR DESCRIPTION
**Pull request by miguelmtzinf**
Closes #316 

Diff of this implementation and the canonical StableDebtToken implementation [here](https://editor.mergely.com/PPPnA0kE). Code can be trimmed (mainly internal functions), but making it revert on mint/burn operations is enough.

### CODE REVIEW

File: src/contracts/facilitators/aave/tokens/GhoStableDebtToken.sol [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+  }
+
+  /**
+   * @notice Calculates the increase in balance since the last user interaction
+   * @param user The address of the user for which the interest is being accumulated
+   * @return The previous principal balance
+   * @return The new principal balance
+   * @return The balance increase
+   */
+  function _calculateBalanceIncrease(
```
Comment (Zer0dot) [inline src/contracts/facilitators/aave/tokens/GhoStableDebtToken.sol:142]: Seems unused, I think we can remove it.

File: src/contracts/facilitators/aave/tokens/GhoStableDebtToken.sol [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    }
+  }
+
+  /**
+   * @notice Burns stable debt tokens of a user
+   * @param account The user getting his debt burned
+   * @param amount The amount being burned
+   * @param oldTotalSupply The total supply before the burning event
+   */
+  function _burn(address account, uint256 amount, uint256 oldTotalSupply) internal {
```
Comment (Zer0dot) [inline src/contracts/facilitators/aave/tokens/GhoStableDebtToken.sol:234]: Seems unused, I think we can remove this too

File: src/contracts/facilitators/aave/tokens/GhoStableDebtToken.sol [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+    return principalSupply.rayMul(cumulatedInterest);
+  }
+
+  /**
+   * @notice Mints stable debt tokens to a user
+   * @param account The account receiving the debt tokens
+   * @param amount The amount being minted
+   * @param oldTotalSupply The total supply before the minting event
+   */
+  function _mint(address account, uint256 amount, uint256 oldTotalSupply) internal {
```
Comment (Zer0dot) [inline src/contracts/facilitators/aave/tokens/GhoStableDebtToken.sol:218]: Seems unused as well, I think it's safe to remove.

**Comment by Zer0dot** [comment]
Overall just a few comments on things I think are safe to remove, though I don't have a strong opinion.

**Comment by miguelmtzinf** [comment]
> Overall just a few comments on things I think are safe to remove, though I don't have a strong opinion. Otherwise, as far as I know, there are no dependencies on the stable debt token for Aave and GHO to function.

No strong opinion neither. I removed them just for bytecode size sake. Looks good to me now

**Pull request merged**